packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -50,9 +50,17 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
 
   final ProductList _history = ProductList.history();
 
+  /// Returns true if the [field] is valid (= not empty).
+  static bool _isProductFieldValid(final String? field) =>
+      field != null && field.trim().isNotEmpty;
+
+  /// Returns true if the [product] basic details are valid (= not empty).
+  static bool isProductBasicValid(final Product product) =>
+      _isProductFieldValid(product.productName) &&
+      _isProductFieldValid(product.brands);
```

Comment:
Looking at it together with @AshAman999 this will probably not work, when you just enter the brand without a name, this will be false so nothing will be saved, see line 134, so you should use a `||` instead of an `??`

and also add the weight into it

```suggestion
  static bool isProductBasicValid(final Product product) =>
      _isProductFieldValid(product.productName) &&
      _isProductFieldValid(product.brands);
```

Reply:
So u want me to remove the validation while adding new product also ?

Reply:
Hello @Sudhanva-Nadiger 

`when you just enter the brand without a name, this will be false so nothing will be saved, see line 134, so you should use a || instead of an ??`

you can do the suggested changes  above

Reply:
Done!